src/components/output/custom-els/MultiPanel/index.ts

```diff
@@ -0,0 +1,208 @@
+
+  constructor () {
+    super();
+
+    // add EventListners
+    this.addEventListener('click', this._onClick, false);
+    this.addEventListener('keydown', this._onKeyDown, false);
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
+  }
+
+  connectedCallback () {
+    this._childrenChange();
+  }
+
+  // Click event handler
+  _onClick(event: Event) {
+    const el: Element = event.target as HTMLElement;
+    this._expand(this._getClosestHeading(el) as HTMLElement);
+  }
+
+  // KeyDown event handler
+  _onKeyDown(event: Event) {
+    const selectedEl = event.target as HTMLElement;
+    const keyboardEvent = event as KeyboardEvent;
+    // if keydown event is not on heading element, ignore
+    if (!this._getClosestHeading(selectedEl)) {
+      return;
+    }
+
+    // don’t handle modifier shortcuts used by assistive technology.
+    if (keyboardEvent.altKey) {
+      return;
+    }
+    let newHeading;
+    switch (keyboardEvent.key) {
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        newHeading = this._prevHeading();
+        break;
+
+      case 'ArrowRight':
+      case 'ArrowDown':
+        newHeading = this._nextHeading();
+        break;
+
+      case 'Home':
+        newHeading = this._firstHeading();
+        break;
+
+      case 'End':
+        newHeading = this._lastHeading();
+        break;
+
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        this._expand(this._getClosestHeading(document.activeElement) as HTMLElement);
+        break;
+
+      // Any other key press is ignored and passed back to the browser.
+      default:
+        return;
+    }
+
+    event.preventDefault();
+    selectedEl.setAttribute('tabindex', '-1');
+    if (newHeading) {
+      newHeading.setAttribute('tabindex', '0');
+      newHeading.focus();
+    }
+  }
+
+  _expand (heading: HTMLElement) {
+
+    if (heading === undefined) {
+      return;
+    }
+
+    const content = heading.nextElementSibling;
+
+    // heading elment should always have nextElementSibling (checked on _childrenChange)
+    // but in case it is null, return.
+    if (content === null) {
+      return;
+    }
+
+    // toggle expanded and aria-expanded attoributes
+    if (content.hasAttribute('expanded')) {
+      content.removeAttribute('expanded');
+      content.setAttribute('aria-expanded', 'false');
+    } else {
+      content.setAttribute('expanded', '');
+      content.setAttribute('aria-expanded', 'true');
+    }
+  }
+
+  // children of multi-panel should always be even nuber (heading/content pair)
+  // if children are odd numbers, add a div at the end to prevent potential error.
+  _childrenChange () {
+    if (this.children.length % 2 !== 0) {
+      console.error(`detected odd number of elements inside multi-panel,
+      please make sure you have heading/content pair`);
+    }
+
+    const children : Element[] = Array.from(this.children);
```

Comment:
Wondering if this loop could use the tree walking approach used elsewhere in this component? It would also make the child count checking easy:

```js
let heading = this.firstElementChild;

while (heading) {
  const content = heading.nextElementSibling;
  if (!content) {
    return console.error('<multi-panel> requires an even number of element children.');
  }

  // L122 - 141 here

  heading = content.nextElementSibling;
}
```